clang/lib/Sema/SemaStmt.cpp

```diff
@@ -302,27 +312,38 @@ void Sema::DiagnoseUnusedExprResult(const Stmt *S, unsigned DiagID) {
     if (const Decl *FD = CE->getCalleeDecl()) {
       if (ShouldSuppress)
         return;
-      if (FD->hasAttr<PureAttr>()) {
+      if (const auto *A = FD->getAttr<PureAttr>()) {
         Diag(Loc, diag::warn_unused_call) << R1 << R2 << "pure";
+        if (OffendingDecl && !OffendingDecl->getIdentifier()->getBuiltinID())
```

Comment:
This situation of an offending decl not being always present is, imo, not really something we should be ok with.  As far as skipping on builtins, I don't know if we SHOULD, I would think the note there is still meaningful.

Reply:
it won't be the 1st line IIRC, the built in location should have the location <builtin>, which makes sense.

I think this test is still sensible though.